### PR TITLE
Issues/399

### PR DESCRIPTION
**Pull request by smcgrath0**
TO Test: 
check base-page under the templates tab for the themeclean-flex site
Should see a reference list to the pages that uses the template (index, error ...)
image

old PR: #403

### CODE REVIEW

**Comment by GastonGonzalez** [comment]
@smcgrath0  - Please resolve this issue in the console.

```
admin.js:33401 [Vue warn]: Duplicate keys detected: '/content/themecleanflex/pages/page1'. This may cause an update error.

found in

---> <AdminComponentsExplorerpreviewcontent>
       <AdminComponentsTemplateview>
         <AdminComponentsWorkspace>
           <AdminComponentsToolingpage>
             <Root>

admin.js:33401 [Vue warn]: Duplicate keys detected: '/content/themecleanflex/pages/page2'. This may cause an update error.

found in

---> <AdminComponentsExplorerpreviewcontent>
       <AdminComponentsTemplateview>
         <AdminComponentsWorkspace>
           <AdminComponentsToolingpage>
             <Root>
```

1. Create two pages.
1. In the second page add a card component and reference the other page.
2. Check the pages references.

File: admin-base/ui.apps/src/main/js/apiImpl.js [hunk 1]
```diff
@@ -471,7 +471,36 @@ class PerAdminImpl {
...
+            //checks if object in position of array is valid
+            if (refBy[i].path){
+              //if count is not set, sets it
+              if (!refBy[i].count){
+                refBy[i].count = 0;
+              }
+              //will give it a count of 1, then if its the different element on the same page, will add again
+              refBy[i].count += 1;
+            }
+            //checks if there is an element next to current element
```
Comment (GastonGonzalez) [inline admin-base/ui.apps/src/main/js/apiImpl.js]: @smcgrath0, consider this: what happens if the response from '/perapi/admin/refBy.json' does not include the `referencedBy.[]` elements sorted by path? Will this code still work?

Comment (GastonGonzalez) [reply admin-base/ui.apps/src/main/js/apiImpl.js]: @smcgrath0 and @dna113p - Why can't we use a `map` and make this dead simple? I am no ES6 expert, but here's what I am thinking. 

```
let referencedBy = [
    {
        "name": "carousel",
        "path": "/content/themecleanflex/pages/library/carousel",
        "propertyName": "imagepath",
        "propertyPath": "/content/themecleanflex/pages/library/carousel/jcr:content/container22/carousel9/slides/c6",
    },
    {
        "name": "sample-owl",
        "path": "/content/themecleanflex/pages/library/carousel/sample-owl",
        "propertyName": "imagepath",
        "propertyPath": "/content/themecleanflex/pages/library/carousel/sample-owl/jcr:content/container175/carousel6/slides/c3",
    },
    {
        "name": "sample-owl",
        "path": "/content/themecleanflex/pages/library/carousel/sample-owl",
        "propertyName": "imagepath",
        "propertyPath": "/content/themecleanflex/pages/library/carousel/sample-owl/jcr:content/container175/carousel12/slides/c9",
    },
]

let result = referencedBy.reduce(
    (map => (r, a) => (!map.has(a.path) && map.set(a.path, r[r.push({ name: a.name, path: a.path, propertyName: a.propertyName, propertyPath: a.propertyPath, count: 0 }) - 1]), map.get(a.path).count++, r))(new Map),
    []
)

console.log(result)
```

It produces this:
```
[
  {
    name: 'carousel',
    path: '/content/themecleanflex/pages/library/carousel',
    propertyName: 'imagepath',
    propertyPath: '/content/themecleanflex/pages/library/carousel/jcr:content/container22/carousel9/slides/c6',
    count: 1
  },
  {
    name: 'sample-owl',
    path: '/content/themecleanflex/pages/library/carousel/sample-owl',
    propertyName: 'imagepath',
    propertyPath: '/content/themecleanflex/pages/library/carousel/sample-owl/jcr:content/container175/carousel6/slides/c3',
    count: 2
  }
]
```

Please let me know if I am way off.

File: admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue [hunk 2]
```diff
@@ -287,7 +287,8 @@
         return this.hasOgTags || this.hasReferences;
       },
       referencedBy() {
-        return $perAdminApp.getView().state.referencedBy.referencedBy
+        let trimmedReferences = this.trimReferences($perAdminApp.getView().state.referencedBy.referencedBy);
+        return trimmedReferences;
```
Comment (GastonGonzalez) [inline admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue]: @smcgrath0  - you never use trimmedReference. Why not just use `return this.trimReferences....`?

File: admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue [hunk 3]
```diff
@@ -370,6 +371,21 @@
...
+            name: a.name,
+            path: a.path,
+            propertyName: a.propertyName,
+            propertyPath: a.propertyPath,
+            count: 0
+          }) - 1]), 
+          map.get(a.path).count++, r))(new Map),
+          []
+        )
+        return result;
```
Comment (GastonGonzalez) [inline admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue]: Same comment as above, @smcgrath0 . get rid of `let result`. You simple use return on referenceLIst.reduc...

**Comment by smcgrath0** [comment]
No review comments, added count ui element
Screen Shot 2020-06-05 at 9 08 50 AM

**Review by GastonGonzalez — CHANGES_REQUESTED** [review]

File: admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue [hunk 4]
```diff
@@ -287,7 +288,8 @@
         return this.hasOgTags || this.hasReferences;
       },
       referencedBy() {
-        return $perAdminApp.getView().state.referencedBy.referencedBy
+        let trimmedReferences = this.trimReferences($perAdminApp.getView().state.referencedBy.referencedBy);
+        return trimmedReferences;
```
Comment (GastonGonzalez) [inline admin-base/ui.apps/src/main/content/jcr_root/apps/admin/components/explorerpreviewcontent/template.vue]: This is var is unused. Just call return.

**Review by GastonGonzalez — CHANGES_REQUESTED** [review]
Please update all changes.

**Review by GastonGonzalez — APPROVED** [review]

**Pull request merged**